magento/magento2
# Guest print order form fixes

### Description (*)

This pull request fixes an issue with guest print order form when a customer inputs data with trailing/leading spaces. The expected result is that spaces are trimmed and order printed successfully.
It also improves Unit tests and Functional tests to cover cases when zip code is used to search for an order on the print guest order page.

### Manual testing scenarios (*)

1. Place an order as a guest
2. Go to the page "sales/guest/form/" on the storefront as a guest.
3. Fill in the print order form:
   -  Try to search for an order by email
   -  Try to search for an order by zip code
   -  Try to input data with trailing/leading spaces

In all cases, you should be able to print the order.

### Related Pull Requests
https://github.com/magento/magento2/pull/30102

### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#29337: Guest print order form fixes
